core/src/com/po/fuck/KeyboardController.java

```diff
@@ -4,58 +4,78 @@
 import com.badlogic.gdx.InputAdapter;
 import com.badlogic.gdx.math.Vector2;
 import com.po.fuck.movement.Boost;
+import com.po.fuck.updates.Updatable;
 
-public class KeyboardController extends InputAdapter {
-    private final Vector2 direction = new Vector2(0, 0);
+public class KeyboardController extends InputAdapter implements Updatable {
     private final Player player;
 
+    // Prevent the situation when the buttons were pressed before the game and did
+    // not get register.
+    private boolean W = false, A = false, S = false, D = false;
+
     KeyboardController(Player player) {
         this.player = player;
     }
 
+    private Vector2 getDirection() {
+        Vector2 result = new Vector2();
+        if (W)
+            result.y -= 1;
+        if (A)
+            result.x -= 1;
+        if (S)
+            result.y += 1;
+        if (D)
+            result.x += 1;
+        return result;
+    }
```

Comment:
Should we scale result Vector2 to the length of `speed`?

Reply:
No, it is only calculating direction